runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -878,4 +887,198 @@ public void shutdown() {
+        @NonNull
+        private final Class<V> vClass;
+        private final Class<M> mClass;
+
+        @Default
+        @Exclude
+        private final boolean withSchema = true;
+
+        @NonNull
+        @Exclude
+        private final TableOptions tableOptions;
+
+        public static <K extends Message, V extends Message, M extends Message>
+        TableDescriptor<K, V, M> build(Class<K> kClass, Class<V> vClass, Class<M> mClass, TableOptions tableOpts) {
+            return TableDescriptor
+                    .<K, V, M>builder()
+                    .kClass(kClass)
+                    .vClass(vClass)
+                    .mClass(mClass)
+                    .withSchema(mClass != null)
+                    .tableOptions(tableOpts)
+                    .build();
+        }
+
+        public static <K extends Message, V extends Message, M extends Message>
+        TableDescriptor<K, V, M> build(Class<K> kClass, Class<V> vClass, Class<M> mClass) {
+            TableOptions defaultOpts;
+            try {
+                defaultOpts = TableOptions.fromProtoSchema(vClass);
+            } catch (Exception e) {
+                throw new IllegalStateException(e);
+            }
+
+            return build(kClass, vClass, mClass, defaultOpts);
+        }
+
+        public SchemaOptions getSchemaOptions() {
+            return tableOptions.getSchemaOptions();
+        }
+
+        public V getDefaultValueMessage() {
+            return getDefaultMessage(vClass);
+        }
+
+        public K getDefaultKeyMessage() {
+            return getDefaultMessage(kClass);
+        }
+
+        private static <T> T getDefaultMessage(Class<T> clazz) {
+            try {
+                return ClassUtils.cast(clazz.getMethod(DEFAULT_INSTANCE_METHOD_NAME).invoke(null));
+            } catch (Exception ex) {
+                throw new IllegalStateException(DEFAULT_METHOD_NOT_FOUND_ERR_MSG);
+            }
+        }
+
+        public void metadataHandler(Consumer<M> defaultMetadataHandler) {
+            if (mClass == null) {
+                return;
+            }
+
+            M defaultMessage = getDefaultMetadataMessage();
+            defaultMetadataHandler.accept(defaultMessage);
+        }
+
+        private M getDefaultMetadataMessage() {
+            if (mClass == null) {
+                return null;
+            }
+
+            return getDefaultMessage(mClass);
+        }
+
+        public Object[] getArgs() {
+            if (withSchema) {
+                return new Object[]{getIndexer()};
+            } else {
+                return new Object[]{};
+            }
+        }
+
+        public ProtobufIndexer getIndexer() {
+            SchemaOptions schemaOptions = getSchemaOptions();
+            V msg = getDefaultValueMessage();
+            return new ProtobufIndexer(msg, schemaOptions);
+        }
+    }
+
+    @Builder
+    @EqualsAndHashCode
+    @ToString
+    public static class FullyQualifiedTableName {
```

Comment:
Is it necessary to expose this externally? I see it helps reduce some code lines internally but from client's usage perspective instead of just using `String` class they have to import `TableRegistry` wherever they are using namespace and tableName right?

Reply:
Yes, think about it, that FullyQualifiedTableName is a protocol we enforce the users to follow, othervise we have sometimes namespace, sometimes it's an empty string sometimes it's a null, you have to use TableRegistry.getFullyQualifiedTableName() method anyway, it's just a bunch of boilerplate code that you have to copy-paste in the end. With my solution you can see through the entire system where FQTableName used and no need for additional boilerplate for streamid and name and other transformation, it get much more easier to use